packages/data/addon/serializers/metadata/bard.ts

```diff
@@ -582,10 +590,14 @@ export default class BardMetadataSerializer extends NaviMetadataSerializer {
         partialData: isNone(description),
         columnFunctionId: metricFunctionId,
       };
-      return this.metricFactory.create(payload);
+      return this.createMetricModel(payload);
     });
   }
 
+  protected createMetricModel(payload: MetricMetadataPayload): MetricMetadataModel {
```

Comment:
It would be nice to have this in the base class and maybe have the factories be private there so both bard/elide serializers only call these functions